core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1611,8 +1611,14 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * Adds a JAR dependency for all tasks to be executed on this SparkContext in the future.
    * The `path` passed can be either a local file, a file in HDFS (or other Hadoop-supported
    * filesystems), an HTTP, HTTPS or FTP URI, or local:/path for a file on every worker node.
+   * If addToCurrentThread is true, attempt to add the new class to the current threads class
+   * loader.
    */
   def addJar(path: String) {
+    addJar(path, false)
+  }
+
+  def addJar(path: String, addToCurrentThread: Boolean) {
```

Comment:
In that case, maybe good to call this out explicitly in the doc?  Also, the naming `addToCurrentThread` makes it seem like it'll get added *only* to the current thread, so maybe something like `addToContextClassLoader` would be more clear.